onnxruntime/test/unittest_main/test_main.cc

```diff
@@ -40,7 +40,11 @@
 #include "core/util/thread_utils.h"
 #include "gtest/gtest.h"
 #include "test/test_environment.h"
-#include "test/framework/TestAllocatorManager.h"
+
+// We keep a list of functions to call on exit before we destroy the OrtEnv.
+// This is needed to do any cleanup that must be done before the OrtEnv gets destroyed and all shared providers get unloaded.
+static std::vector<std::function<void()>> exit_functions;
```

Comment:
Does it make our unit test program too complicated?

Reply:
If unit tests need special handlings like this, probably end users need such things too. It doesn't look good.

Reply:
I don't think so, but since we love using global variables, we need a way to clean things up before we destroy the OrtEnv.